cmake/SetupCppCheck.cmake

```diff
@@ -6,19 +6,41 @@ if(NOT CPPCHECK_FOUND AND NOT CPPCHECK_EXECUTABLE)
 endif()
 
 if(CPPCHECK_EXECUTABLE)
+  # The reason for checking if the printed out file is empty is because
+  # cppcheck check returns 0 even if some checks fail. If the
+  # --error-exitcode flag is set, then it always returns the error
+  # code.
```

Comment:
Because cppcheck has that option and so if someone looks at what cppcheck offers they'd be confused why we aren't using that flag. This comment is designed to explain that

Reply:
Then can I suggest "We do not use cppcheck's `--error-exitcode` flag, because this makes cppcheck always return the error code"

Reply:
sounds good to me